bluesky/run_engine.py

```diff
@@ -1212,12 +1309,21 @@ def _stage(self, msg):
 
     @asyncio.coroutine
     def _unstage(self, msg):
+        """Instruct the RunEngine to unstage the object
+
+        Expected message object is:
+
+            Msg('unstage', object)
+        """
         _, obj, args, kwargs = msg
         # If an object has no 'unstage' method, assume there is nothing to do.
         if not hasattr(obj, 'unstage'):
             return
         result = obj.unstage()
```

Comment:
Is it worth warning the user if the object is not in the set?  Something like:

```python
        if obj not in self._staged:
            self.log.warn('obj: %s is not in the set of staged objects', obj)
        self._staged.discard(obj)
```
or
```python
        try:
            self._staged.discard(obj)
        except KeyError as ke:
            self.log.warn('obj: %s is not in the set of staged objects', obj)
```

or just silently eat the discard:

```python
self._staged.discard(obj)
```

Reason for not just silently eating the discard would be if someone writing a custom scan wanted to eke out a little more performance by not calling unstage on objects that were never staged, but I'm not even convinced that is a use-case worth supporting.  Probably simpler to just do the third option above and leave it at that. Thoughts?